### PR TITLE
CP-36113: Imported the SDK repo

### PR DESCRIPTION
**Pull request by kc284**
The last two commits (8cc7111 and 1e8b0ab) contain new, unreviewed code; everything else was reviewed in the SDK repo.

### CODE REVIEW

**Comment by robhoes** [comment]
Welcome back SDK!

**Comment by psafont** [comment]
Nice! could you add the `xen-api-sdk` package to the tests? https://github.com/xapi-project/xen-api/blob/master/.github/workflows/main.yml#L28

File: Makefile [hunk 1]
```diff
@@ -32,6 +33,32 @@ doc:
...
+	mkdir -p $(XAPISDK)/powershell
+	mkdir -p $(XAPISDK)/python/samples
+	cp -r _build/default/ocaml/sdk-gen/c/autogen/* $(XAPISDK)/c
+	cp -r _build/default/ocaml/sdk-gen/csharp/autogen/* $(XAPISDK)/csharp
+	cp -r _build/default/ocaml/sdk-gen/java/autogen/* $(XAPISDK)/java
+	cp -r _build/default/ocaml/sdk-gen/powershell/autogen/* $(XAPISDK)/powershell
+	cp -r ocaml/sdk-gen/python/samples/*.py $(XAPISDK)/python/samples
+	cp scripts/examples/python/XenAPI/XenAPI.py $(XAPISDK)/python
+	sh ocaml/sdk-gen/windows-line-endings.sh $(XAPISDK)/csharp
+	sh ocaml/sdk-gen/windows-line-endings.sh $(XAPISDK)/powershell
```
Comment (robhoes) [inline Makefile:60]: This looks like something we eventually could get dune to do? I should be possible to get dune to automatically put the installation targets into the right location under `_build/install`.

Comment (kc284) [reply Makefile:60]: The install stanza does not seem to support multiple files/directories yet, this issue is still open: https://github.com/ocaml/dune/issues/1868. I also tried the sites stanza, but it needs a higher dune version. Any other suggestions?

Comment (robhoes) [reply Makefile:60]: Hmm... I don't think its the install stanza you need here, which is for installing files into the wider filesystem. The build command puts the build targets under `_build/install`. So I think these files need to be marked as targets somehow (I'm probably getting the terminology wrong). I'll take a look as well.

Comment (robhoes) [reply Makefile:60]: Oh, I see now that it is indeed the install "stanza" that you need, which is implicit when your use `public_name` in `executable` and things like that.

Comment (robhoes) [reply Makefile:60]: Anyway, let's leave it like this for now (we'll have something similar for `make doc`).

**Review by robhoes — APPROVED** [review]

**Comment by psafont** [comment]
The first check complains about formatting. This can be fixed by installing ocamlformat using `opam install ocmalformat` in an opam switch with xs-opam as default repo and running make format.

I could make the change and push it, if you like.

**Review by psafont — APPROVED** [review]
The DCO failing it's unfortunate but I would rather merge the history without changing it.

**Comment by robhoes** [comment]
Yes, we'll ignore the DCO check for this kind of thing.

**Pull request merged**